### PR TITLE
feat: allow for registry override

### PR DESCRIPTION
**Pull request by bee344**
### Description
Updated `parseRegistry` to allow for overriding entries of the existing registry. This is achieved by adding the optional property `overridingRegistry` to `AssetTransferApiOpts`. Now `propertyIterator` takes an extra optional argument `override`, a `boolean` that defines whether we intend to override the registry or not. This value is set on `parseRegistry` and depends on the presence of the property `overridingRegistry` in the `opts` passed to the function. If the `chainId` passed in `overridingRegistry` is not present in the registry, it just adds it as if it were an `injectedRegistry`. If both options are present, it first updates the registry with `injectedRegistry` and then makes the changes as per `overridingRegistry`.

### CODE REVIEW

File: src/registry/parseRegistry.spec.ts [hunk 1]
```diff
@@ -469,73 +483,287 @@ describe('parseRegistry', () => {
...
+						tokens: ['WOP', 'WER'],
+					},
+				},
+			},
+		};
+		const registry = parseRegistry(reg as ChainInfoRegistry<ChainInfoKeys>, opts);
+
+		expect(registry.westend['0'].tokens).toStrictEqual(['WOP', 'WER']);
+	});
+	it("Should correctly override the registry's token entry", () => {
```
Comment (marshacb) [inline src/registry/parseRegistry.spec.ts:570]: duplicate test case

**Review by marshacb — APPROVED** [review]
minor nit. LGTM

File: src/types.ts [hunk 2]
```diff
@@ -154,6 +154,10 @@ export type AssetTransferApiOpts<T extends ChainInfoKeys | InjectedChainInfoKeys
 	 * Option to inject chain information into the registry.
 	 */
 	injectedRegistry?: RequireAtLeastOne<ChainInfoRegistry<T>>;
+	/**
+	 * Option to override the registry with the supplied chain information.
+	 */
+	overridingRegistry?: RequireAtLeastOne<ChainInfoRegistry<InjectedChainInfoKeys>>;
```
Comment (TarikGul) [inline src/types.ts:160]: To keep things in the active tense, I think we should call this `overrideRegistry`.

**Review by TarikGul — CHANGES_REQUESTED** [review]
Overall looks amazing! Small nit, I think we should change the name to `overrideRegistry`. In addition to that we need to update the docs in the README.md that go over the `AssetTransferApiOpts`.

**Review by TarikGul — APPROVED** [review]
LGTM, great job :)

**Comment by Polkadot-Forum** [comment]
This pull request has been mentioned on **Polkadot Forum**. There might be relevant details there:

https://forum.polkadot.network/t/parity-tech-update-for-february/6630/1

**Pull request merged**